src/fireedge/src/modules/components/Forms/VrTemplate/InstantiateForm/Steps/UserInputs/Node/UserInputDialog.js

```diff
@@ -0,0 +1,176 @@
+    },
+  })
+
+  useEffect(() => {
+    setValue(`user_inputs.${userInput.name}`, userInput?.default || '')
+  }, [userInput, setValue])
+
+  const onSubmit = () => {
+    onClose()
+  }
+
+  const renderInputField = ({ field }) => {
+    const handleNumericChange = (event) => {
+      const value = event.target.value
+      field.onChange(userInput.type === 'number' ? Number(value) : value)
+    }
+    switch (userInput.type) {
+      case 'list':
+      case 'listmultiple':
+        return (
+          <FormControl fullWidth>
+            <InputLabel id="value-select-label">Value</InputLabel>
+            <Select
+              {...field}
+              labelId="value-select-label"
+              id="value-select"
+              label={Tr(T.Value)}
+              multiple={userInput?.type === 'listmultiple'}
+            >
+              {userInput?.options?.map((option, index) => (
+                <MenuItem key={index} value={option}>
+                  {option}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        )
+      case 'boolean':
+        return (
+          <RadioGroup
+            row
+            aria-labelledby="boolean-choice-label"
+            name="boolean-choice"
+            value={field.value}
+            onChange={(e) => field.onChange(e.target.value)}
+          >
+            <FormControlLabel
+              value="YES"
+              control={<Radio />}
+              label={Tr(T.Yes)}
+            />
+            <FormControlLabel value="NO" control={<Radio />} label={Tr(T.No)} />
+          </RadioGroup>
+        )
+      default:
+        return (
+          <TextField
+            {...field}
+            margin="dense"
+            id="value"
+            label={Tr(T.Value)}
+            type={userInput?.type === 'number' ? 'number' : 'text'}
+            fullWidth
+            variant="outlined"
+            onChange={
+              userInput.type === 'number' ? handleNumericChange : field.onChange
+            }
+          />
+        )
+    }
+  }
+
+  const handleDialogClick = (event) => {
+    event.stopPropagation()
+  }
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      aria-labelledby="form-dialog-title"
+      onClick={handleDialogClick}
+    >
+      <form
+        onSubmit={(e) => {
+          e.preventDefault()
+          onSubmit()
+        }}
+      >
+        <DialogTitle id="form-dialog-title">{userInput.name}</DialogTitle>
+        <DialogContent>
+          {userInput.description && (
+            <Typography style={{ marginBottom: '16px' }}>
+              {userInput.description}
+            </Typography>
+          )}
+          <Controller
+            name={`user_inputs.${userInput.name}`}
+            control={control}
+            render={renderInputField}
+          />
+        </DialogContent>
+        <DialogActions>
+          <Button type="submit" color="primary">
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX element content is not internationalized. Application should support multiple languages through internationalization. Instead, utilize packages like &#39;i18next&#39; to internationalize your items.